asyncscope.md

```diff
@@ -279,201 +352,541 @@ listening activity is bound in the scope of the loop, the lifetime of handling r
+the scope has transitioned to the closed/joining state since the _`nest-sender`_ was constructed, which means that there
+is a _`join-sender`_ waiting to complete so, if the decrement brings the count of outstanding senders to zero then the
+waiting _`join-sender`_ needs to be notified that the scope is now joined and the sender can complete.
+
+A call to `nest()` does not start the given sender. A call to `nest()` is not expected to incur allocations other than
+whatever might be required to move or copy `s`.
 
 Similar to `spawn_future()`, `nest()` doesn't constrain the input sender to any specific shape. Any type of sender is
 accepted.
 
-Unlike `spawn_future()` the returned sender does not prevent the scope from ending. It is safe to drop the returned
-sender without starting it. It is UB to start the returned sender after the `counting_scope` has been destroyed.
-
 As `nest()` does not immediately start the given work, it is ok to pass in blocking senders.
 
-One can say that `nest()` is more fundamental than `spawn()` and `spawn_future()` as the latter two can be implemented
-in terms of `nest()`. In terms of performance, `nest()` does not introduce any penalty. `spawn()` is more expensive than
-`nest()` as it needs to allocate memory for the operation. `spawn_future()` is even more expensive than `spawn()`; the
-receiver needs to be type-erased and a possible race condition needs to be resolved. `nest()` does not require
-allocations, so it can be used in a free-standing environment.
-
-Cancelling the returned sender, once it is connected and started, cancels `s` but does not cancel the `counting_scope`.
+`nest()` is lvalue-ref qualified as it would be inappropriate to nest senders in a temporary---the temporary's
+destructor would unconditionally invoke `std::terminate()` as there would be no way to move it into the joined state.
 
 Usage example:
-```c++
+```cpp
 ...
 sender auto snd = s.nest(key_work());
-for ( int i=0; i<10; i++)
+for (int i = 0; i < 10; i++)
     spawn(s, on(sched, other_work(i)));
 return on(sched, std::move(snd));
 ```
 
+### `counting_scope::join()`
+
+```cpp
+struct @@_join-sender_@@; // @@_exposition-only_@@
+
+[[nodiscard]] @@_join-sender_@@ join() noexcept;
+```
+
+Returns a _`join-sender`_. When the _`join-sender`_ is connected to a receiver, `r`, it produces an
+_`operation-state`_, `o`. When `o` is started, the scope moves from the open state to the closed/joining state. `o`
+completes with `set_value()` when the scope moves from the closed/joining state to the closed state, which happens when
+the scope's count of outstanding senders drops to zero. `o` may complete synchronously if it happens to observe that the
+count of outstanding senders is already zero when started; otherwise, `o` completes on the execution context it was
+started on by asking its receiver, `r`, for a scheduler, `sch`, with `get_scheduler(get_env(r))` and then starting the
+sender returned from `schedule(sch)`.
```

Comment:
this implies that i cannot connect a join sender to a receiver that doesn't have a scheduler in its environment. we should probably add justification for that design decision.